src/Core/Compiler/TemplateCompiler.php

```diff
@@ -37,13 +39,49 @@ class TemplateCompiler {
 	protected $renderingContext;
 
 	/**
+	 * @var string
+	 */
+	protected $mode = self::MODE_NORMAL;
```

Comment:
Hmm, if I remember correctly this was first supported in PHP 4. But must admit my memory isn't that good. It is *certainly* supported on all our supported versions ;)